roles/reboot/tasks/main.yml

```diff
@@ -29,18 +44,20 @@
 # that they don't use sudo, which may require a password, and is not necessary
 # anyway.
 
-- name: wait for hosts to go down
-  local_action:
-    wait_for host={{ real_ansible_host }}
-    port=22 state=absent delay=1 timeout={{ timeout }}
-  become: false
-
 - name: wait for hosts to come back up
   local_action:
     wait_for host={{ real_ansible_host }}
-    port=22 state=started delay=30 timeout={{ timeout }}
+    port={{ real_ansible_port | default('22') }} state=started delay={{ wf_delay }} timeout={{ timeout }}
   become: false
 
+# I'm not sure the retries are even necessary, but I'm keeping them in
```

Comment:
I started coding this too a bit before but I think the problem is we actually need to retry *both* the ssh connection and the boot_id check.  The way to think of this is: it was still racy before the STR approach since we were relying on the shutdown actually finding the ssh port closed, but we could (if the guest rebooted fast enough) have it always appear open. So we need to loop over both tasks.  This led me to https://stackoverflow.com/questions/30785281/one-loop-over-multiple-ansible-tasks

Reply:
@cgwalters Did you have to do this in any of the ostree tests you've been working on?

What's the idea for the implementation?  Break out those tasks into a separate file and loop over the `include` of the file?

Reply:
@mike-nguyen found this post which discusses the problem of rebooting a Vagrant box and the SSH port is forwarded to a non-standard port (much like the STI does):

https://blog.jeffli.me/blog/2017/05/19/a-pitfall-of-vagrant-when-using-ansibles-wait-for-module/

TL;DR - use `search_regex` with the `wait_for` module

I'm going to try to play with this approach sometime today.